main.py

```
@@ -105,3 +126,33 @@ def create_personality(personality: schemas.PersonalityCreate, db: Session = Dep
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Arquivo não encontrado com esta chave.")
+
+    # Simulação da resposta
+    return {"message": f"Arquivo para a chave '{retrieval_key}' seria entregue aqui.", "product_name": product.name}
+
+@app.post("/upload-products", tags=["Products"])
+async def upload_products(tenant_id: str, file: UploadFile, db: Session = Depends(get_db)):
+    try:
+        products = await services.process_product_sheet(tenant_id, file, db)
+        return {"message": f"{len(products)} products uploaded successfully."}
+    except Exception as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)